rotkehlchen/data_import/importers/_import_helpers.py

```diff
@@ -0,0 +1,241 @@
+
+
+def is_withdrawal(
+    transaction_type: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+) -> bool:
+    """Check if the transaction is a withdrawal."""
+    if not is_same_asset(origin_asset, destination_asset):
+        raise ValueError(
+            'Transaction Type and origin/destination assets don\'t match. '
+            f'Transaction Type is {transaction_type} but origin asset is {origin_asset} '
+            f'and destination asset is {destination_asset}',
+        )
+    # Add negative check to serve for 'is_deposit'
+    return transaction_type == 'out' and transaction_type != 'in'
+
+
+def is_deposit(
+    transaction_type: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+) -> bool:
+    """Check if the transaction is a deposit."""
+    return not is_withdrawal(transaction_type, origin_asset, destination_asset)
+
+
+def is_trade(origin_asset: Asset, destination_asset: Asset) -> bool:
+    """Check if the transaction is a trade."""
+    return not is_same_asset(origin_asset, destination_asset)
+
+
+def is_ledger_action(
+    origin_asset: Asset,
+    destination_asset: Asset,
+    origin_amount: AssetAmount,
+    destination_amount: AssetAmount,
+) -> bool:
+    """Check if the transaction is a ledger action."""
+    return origin_asset == destination_asset and origin_amount == destination_amount
+
+
+def get_action_type(transaction_type: str) -> LedgerActionType | None:
+    if transaction_type == 'in':
+        return LedgerActionType.INCOME
+    if transaction_type == 'out':
+        return LedgerActionType.EXPENSE
+    log.debug(f'Ignoring uncaught transaction type of {transaction_type}.')
+    return None
+
+
+class TransactionType:
+    class AssetMovement:
+        def __call__(
+            self,
+            location: Location,
+            category: AssetMovementCategory,
+            timestamp: Timestamp,
+            asset: Asset,
+            amount: AssetAmount,
+            fee: Fee,
+            fee_asset: Asset,
+            link: str = '',
+            address: str = None,
+            transaction_id: str = None,
+        ) -> AssetMovement:
+            return AssetMovement(
+                location=location,
+                category=category,
+                address=address,
+                transaction_id=transaction_id,
+                timestamp=timestamp,
+                asset=asset,
+                amount=amount,
+                fee=fee,
+                fee_asset=fee_asset,
+                link=link,
+            )
+
+    class Trade:
+        def __call__(
+            self,
+            location: Location,
+            fee_asset: Asset,
+            quote_asset: Asset,
+            destination_amount: AssetAmount,
+            trade_type: TradeType,
+            fee: Fee,
+            notes: str,
+            origin_amount: AssetAmount,
+            base_asset: Asset,
+            timestamp: Timestamp,
+            link: str = '',
+        ) -> Trade | None:
```

Comment:
We use `typing` module in our codebase, so I guess `Union[Optional[Trade]]` will be better